revocation/revocation.go

```diff
@@ -0,0 +1,122 @@
+// Package Revocation provides methods for checking the revocation status of a certificate chain
+package revocation
+
+import (
+	"crypto/x509"
+	"net/http"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/revocation/ocsp"
+)
+
+// Revocation is an interface that specifies methods used for revocation checking
+type Revocation interface {
+	Validate(certChain []*x509.Certificate, signingTime time.Time) error
+	OCSPStatus(certChain []*x509.Certificate, signingTime time.Time) error
+	SetMergeErrorsFunction(mergeErrorsFunc func(errors []error) error)
+}
+
+// revocation is an internal struct used for revocation checking
+type revocation struct {
+	httpClient      *http.Client
+	mergeErrorsFunc func(errors []error) error
+}
+
+// New constructs a revocation object and substitutes default values for any that are passed as nil
+func New(httpClient *http.Client) Revocation {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+	return &revocation{
+		httpClient:      httpClient,
+		mergeErrorsFunc: DefaultMergeErrors,
+	}
+}
+
+// Validate checks OCSP, returns nil if all certs in the chain are not revoked.
+// If there is an error, it will return one of the errors defined in the revocation/ocsp package in errors.go.
+// (e.g. if a certificate in the chain is revoked by OCSP and there are no other errors, it will return ocsp.RevokedError)
+//
+// TODO: add CRL support
+func (r *revocation) Validate(certChain []*x509.Certificate, signingTime time.Time) error {
+	return r.OCSPStatus(certChain, signingTime)
+	// TODO: add CRL support
+	// https://github.com/notaryproject/notation-core-go/issues/125
+}
+
+// OCSPStatus checks OCSP, returns nil if all certs in the chain are not revoked.
+// If there is an error, it will return one of the errors defined in the revocation/ocsp package in errors.go.
+// (e.g. if a certificate in the chain is revoked by OCSP and there are no other errors, it will return ocsp.RevokedError)
+func (r *revocation) OCSPStatus(certChain []*x509.Certificate, signingTime time.Time) error {
```

Comment:
We can call it directly within the validate function, but this would force the users to use our method of validation rather than being able to perform OCSP/CRL validation separately or in a different order (which would become even more of an issue if ocsp is moved to internal). IMO it is harmful to the user experience to restrict the way they can perform revocation checks